model/ServerHandler.php

```diff
@@ -0,0 +1,151 @@
+           '_can_delete'    => pacol(   0,        0, 1, 'int',  '',              '', 0, array(), 0, 1, (Config::bool('multiple_servers') && $this->is_superadmin) . ' as _can_delete' ),
+        );
+    }
+
+    protected function initMsg() {
+        $this->msg['error_already_exists'] = 'server_already_exists';
+        $this->msg['error_does_not_exist'] = 'server_does_not_exist';
+        $this->msg['confim_delete'] = 'confirm_delete_server';
+        if ($this->new) {
+            $this->msg['logname'] = 'create_server';
+            $this->msg['store_error'] = 'server_create_error';
+            $this->msg['successmessage'] = 'server_store_success';
+        } else {
+            $this->msg['logname'] = 'edit_server';
+            $this->msg['store_error'] = 'server_edit_error';
+            $this->msg['successmessage'] = 'server_updated';
+        }
+        $this->msg['can_create'] = Config::bool('multiple_servers') && $this->is_superadmin;
+    }
+
+    public function webformConfig() {
+        return array(
+            'formtitle_create' => 'server_creation',
+            'formtitle_edit' => 'server_edit',
+            'create_button' => 'server_button',
+
+            'required_role' => 'global-admin',
+            'listview' => 'list.php?table=server',
+            'early_init' => 0,
+        );
+    }
+
+    protected function storemore() {
+        if ($this->new) {
+            if (!$this->server_postcreation()) {
+                $this->errormsg[] = Config::lang('server_postcreate_failed');
+            }
+        }
+        return true;
+    }
+
+    protected function server_postcreation() {
+        $script=Config::read_string('server_postcreation_script');
+
+        if (empty($script)) {
+            return true;
+        }
+
+        if (empty($this->id)) {
+            $this->errormsg[] = 'Empty server parameter in server_postcreation';
+            return false;
+        }
+
+        $cmdarg1=escapeshellarg($this->id);
+        $command= "$script $cmdarg1";
+        $retval=0;
+        $output=array();
+        $firstline='';
+        $firstline=exec($command,$output,$retval);
+        if (0!=$retval) {
+            error_log("Running $command yielded return value=$retval, first line of output=$firstline");
+            $this->errormsg[] = 'Problems running server postcreation script!';
+            return false;
+        }
+
+        return true;
+    }
+
+    protected function server_postdeletion() {
+        $script=Config::read_string('server_postdeletion_script');
+
+        if (empty($script)) {
+            return true;
+        }
+
+        if (empty($this->id)) {
+        }
+
+        $cmdarg1=escapeshellarg($this->id);
+        $command= "$script $cmdarg1";
+        $retval=0;
+        $output=array();
+        $firstline='';
+        $firstline=exec($command,$output,$retval);
+        if (0!=$retval) {
+            error_log("Running $command yielded return value=$retval, first line of output=$firstline");
```

Comment:
Why not (for example) just json_encode($output) at the end of this message, rather than restricting it to the first line ?

Reply:
To be honest, I've just copied this piece of code from [model/DomainHandler.php#L292](https://github.com/postfixadmin/postfixadmin/blob/e8f27969a3a1be7461efc3ece169605fec4c5f43/model/DomainHandler.php#L292)

I could change that in my pull request, but I guess it should then be changed in all model handlers.